static/app/views/usageStats/utils.tsx

```diff
@@ -102,3 +109,34 @@ export function isDisplayUtc(datetime: DateTimeObject): boolean {
   const hours = parsePeriodToHours(interval);
   return hours >= 24;
 }
+
+/**
+ * Duplicate of getInterval from static/app/components/charts/utils
+ * except that we do not support <1h granularity for time ranges >6h
+ */
+export function getUsageInterval(datetimeObj: DateTimeObject) {
+  const diffInMinutes = getDiffInMinutes(datetimeObj);
+
+  if (diffInMinutes >= SIXTY_DAYS) {
+    // Greater than or equal to 60 days
+    return '1d';
+  }
+
+  if (diffInMinutes >= THIRTY_DAYS) {
+    // Greater than or equal to 30 days
+    return '4h';
+  }
+
+  if (diffInMinutes > ONE_HOUR * 6) {
+    // Between 6 hour and 30 days
+    return '1h';
+  }
+
+  if (diffInMinutes > ONE_HOUR) {
```

Comment:
we should probably not allow any intervals below an hour even for these small ranges, and in the future enforce that on the backend (unless someone asks for this feature) based on our slack discussion. is it possible to limit interval selection?

Reply:
😢